src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -1253,18 +1242,29 @@ private TextLoader(IHost host, ModelLoadContext ctx, IMultiStreamSource files)
             _parser = new Parser(this);
         }
 
-        public static TextLoader Create(IHostEnvironment env, ModelLoadContext ctx, IMultiStreamSource files)
+        public static TextLoader Create(IHostEnvironment env, ModelLoadContext ctx)
         {
             Contracts.CheckValue(env, nameof(env));
             IHost h = env.Register(RegistrationName);
 
             h.CheckValue(ctx, nameof(ctx));
             ctx.CheckAtModel(GetVersionInfo());
-            h.CheckValue(files, nameof(files));
 
-            return h.Apply("Loading Model", ch => new TextLoader(h, ctx, files));
+            return h.Apply("Loading Model", ch => new TextLoader(h, ctx));
         }
 
+        // These are legacy constructors needed for ComponentCatalog.
+        public static IDataLoader Create(IHostEnvironment env, ModelLoadContext ctx, IMultiStreamSource files)
+            => (IDataLoader)Create(env, ctx).Read(files);
+        public static IDataLoader Create(IHostEnvironment env, Arguments args, IMultiStreamSource files)
+            => (IDataLoader)new TextLoader(env, args, files).Read(files);
+
+        /// <summary>
+        /// Convenience method to read a text data file into a <see cref="IDataView"/>.
+        /// </summary>
+        public static IDataView ReadFile(IHostEnvironment env, Arguments args, IMultiStreamSource fileSource)
```

Comment:
>ReadFile [](start = 32, length = 8)

We can surely wordsmith this better as we move on, but the phrasing here suggests that it will actually be materialized into memory (that is, my expectation reading the docs above is that the text file *will* be read by this method), which is not the case of course.